cryptoki/src/functions/slot_token_management.rs

```diff
@@ -47,6 +47,27 @@ impl Pkcs11 {
         Ok(slots)
     }
 
+    /// Get all slots available with a token
+    pub fn get_slots_with_initialized_token(&self) -> Result<Vec<Slot>> {
+        let slots = self.get_slots_with_token()?;
+
+        let slots: Vec<Slot> = slots
+            .into_iter()
+            .filter_map(|slot| match self.get_token_info(slot) {
+                Ok(token_info) => {
+                    if token_info.get_flags().token_initialized() {
+                        Some(Ok(slot))
+                    } else {
+                        None
+                    }
+                }
+                Err(e) => Some(Err(e)),
+            })
+            .collect::<Result<Vec<Slot>>>()?;
```

Comment:
This is minor but you can either omit type declaration in `let slots: Vec<Slot>` *or* shorten the explicit type in collect to `collect::<Result<_>>()?;`

Reply:
Since this function returns a `Result<Vec<Slot>>, I am wondering if you could also remove everything:
- the `let slots: Vec<Slot> =` at the beginning
- just have `.collect()` at the end (without `?` and `;`)

Reply:
Indeed. That'd make it more readable too IMHO.